src/services/createRouter.ts

```diff
@@ -124,7 +111,7 @@ export function createRouter<
     const to = find(url, options) ?? getRejectionRoute('NotFound')
     const from = { ...currentRoute }
 
-    const beforeResponse = await runBeforeRouteHooks({ to, from, hooks })
+    const beforeResponse = await hooks.runBeforeRouteHooks({ to, from })
```

Comment:
I really like that since the `createRouterHooks` service has the hooks store internally, we don't have to pass it around.